distsql/request_builder.go

```diff
@@ -715,6 +715,8 @@ func encodeIndexKey(sc *stmtctx.StatementContext, ran *ranger.Range) ([]byte, []
 		}
 	}
 
+	// NOTE: this is a hard-code operation to avoid wrong results when accessing unique index with NULL;
+	// Please see https://github.com/pingcap/tidb/issues/29650 for more details
```

Comment:
Will we remove the hard-code operation in the following PRs?

Reply:
Yep, I plan to remove this hard-code operation with the switch `tidb_regard_null_as_point` together a few months later if there is no new bug.